### PR TITLE
[PD-6476] - location url and location input format

### PR DESCRIPTION
**Pull request by jeremytubbs**
- Adds new `/locations/:location/jobs` endpoint - http://localhost:8080/locations/flower-mound-tx/jobs
- Adds a helper to format the location input from a slug

### CODE REVIEW

File: src/components/Search/AppSearchProvider.vue [hunk 1]
```diff
@@ -172,11 +173,23 @@ export default {
...
+
         setInputFromQuery() {
             this.input = merge(this.getInputDefaults(), this.$route.query)
-
             this.formatInput()
         },
+
+        setInputFromParams (query) {
+            let params = { ... this.$route.params }
+            Object.entries(params).map(([key, value]) => {
```
Comment (aljordan82) [inline src/components/Search/AppSearchProvider.vue:187]: I don't think Object.entries is supported in IE11, 
are you using a polyfill?

Comment (jeremytubbs) [reply src/components/Search/AppSearchProvider.vue:187]: According to https://caniuse.com/object-entries a polyfill is included in core-js which is in our dependencies https://github.com/zloirock/core-js#ecmascript-object

Comment (aljordan82) [reply src/components/Search/AppSearchProvider.vue:187]: There shouldn't be a need to import it if corejs is part of the build.

Comment (jeremytubbs) [reply src/components/Search/AppSearchProvider.vue:187]: Yeah, I believe it is included via the `@vue/cli-plugin-babel/preset` defined in `babel.config.js`

File: src/components/Search/AppSearchProvider.vue [hunk 2]
```diff
@@ -172,11 +173,23 @@ export default {
...
+            this.input.location = this.input.location ? displayLocationFromSlug(this.input.location) : ""
+        },
+
         setInputFromQuery() {
             this.input = merge(this.getInputDefaults(), this.$route.query)
-
             this.formatInput()
         },
+
+        setInputFromParams (query) {
```
Comment (surgiie) [inline src/components/Search/AppSearchProvider.vue:185]: `query` is not being used anywhere in the function

File: src/services/helpers.js [hunk 3]
```diff
@@ -69,6 +69,12 @@ export function buildJobDetailUrl(title, location, guid) {
...
 const slugify = (string) => (
     words(toString(string).replace(/["\u2019+:]/g, ""), /[\w]+/g).reduce((result, word, index) => (
-      result + (index ? "-" : "") + word.toLowerCase()
+        result + (index ? "-" : "") + word.toLowerCase()
     ), "")
-  )
\ No newline at end of file
+)
+
+export function displayLocationFromSlug(string) {
```
Comment (surgiie) [inline src/services/helpers.js:76]: Im assuming at some point we will be undoing slugs for more than just location so i think we can make a more generic function for this maybe one that looks unslugifies (is that a word?) and accepts a delimiter for in between words?

ex `unslugify(location, ",")`

File: src/components/Search/AppSearchProvider.vue (hunk 2 above)
Comment (jeremytubbs) [reply src/components/Search/AppSearchProvider.vue]: removed

**Review by surgiie — APPROVED** [review]

File: src/services/helpers.js (hunk 3 above)
Comment (jeremytubbs) [reply src/services/helpers.js:76]: No this is also capitalizes the last... and hopefully I will not need to do this anywhere else.

**Pull request merged**